monai/data/image_reader.py

```diff
@@ -780,13 +787,22 @@ def _extract_region(
         level: int = 0,
         dtype: DtypeLike = np.uint8,
     ):
-        # reverse the order of dimensions for size and location to be compatible with image shape
-        location = location[::-1]
-        if size is None:
-            region = img_obj.read_region(location=location, level=level)
+        if self.backend == "tifffile":
+            region = img_obj.asarray(level=level)
+            img_obj.close()
```

Comment:
Maybe it's better to close the `img_obj` at the same place it opened, instead of this `extract` function?

Thanks.

Reply:
I agree with you @Nic-Ma but the logic of image reader does not allow this since the object is being created at `.read` method and but used in `.get_data`. So here I am closing it as soon as possible. Thanks

Reply:
Is there a way to check whether the image is opened or not?
So if users call `get_data()` multiple times, they should be notified to call `read()` first.

Thanks.

Reply:
@drbeh I use tiffile to read image simply as
` image = tifffile.imread(image_path, key=self.level)`

do we need to call close() at all?

Reply:
@myron you're right but the way classes inherited from `ImageReader` work is to use `read()` to read image object and then use `get_data()` to retrieve the image, so instead of `imread`, we should use `TiffFile` which requires to be closed.

Reply:
@Nic-Ma I checked on this issue. The TiffFile object is being opened where `.asarray()` is being called so the implementation is actually fine. However, to avoid confusion I used context manager to when calling `asarray`.
I also updated a test to include multiple calls to `get_data` to make sure that there is not any problem with closing this file.

Reply:
looks good to me